server/src/test/java/com/broadcom/lsp/cobol/service/ClientServerIntegrationTest.java

```diff
@@ -0,0 +1,177 @@
+  private static final String COPYBOOK1 = "       COPY CPYBK1.";
+  private static final String COPYBOOK2 = "       COPY CPYBK2.";
+  private static final String TEXT =
+      "       Identification Division. \n"
+          + "       Program-id.    ProgramId.\n"
+          + "       Data Division.\n"
+          + "       Working-Storage Section.\n"
+          + "       01   outer1. \n"
+          + "        02   inner1      PIC 9(4) Binary. \n"
+          + "        02   inner2      PIC X(10).\n"
+          + "       01   outer2.\n"
+          + "        02   inner1      PIC 9(4) Binary. \n"
+          + "        02   inner2      PIC X(10).\n"
+          + "       02   Str          PIC 9(4) Binary. \n"
+          + "       Procedure Division.\n"
+          + "       000-Main-Logic.\n"
+          + "           Perform 100-Test.\n"
+          + "           Stop Run.\n"
+          + "       100-Test.\n"
+          + "           Move inner1 of outer1 to Str.\n"
+          + "           Move inner2 of outer1 to Str.\n"
+          + "           Move inner1 of outer2 to Str.\n"
+          + "           Move inner2 of outer2 to Str.\n"
+          + "       COPY CPYBK1.\n"
+          + "       COPY CPYBK2.\n"
+          + "       COPY CPYBK1.\n"
+          + "       End program ProgramId.";
+  @Inject TextDocumentService service;
+  @Inject LanguageServer server;
+  @Inject TestLanguageClient client;
+  @Inject CopybookService copybookService;
+
+  /**
+   * This method tests that after a shutdown request, {@link TextDocumentService} always return
+   * SHUTDOWN_RESPONSE
+   *
+   * @throws ExecutionException
+   * @throws InterruptedException
+   */
+  @Test
+  void whenShutdownIsFired_ThenNewRequestReturnInvalidResponse()
+      throws ExecutionException, InterruptedException {
+
+    server.shutdown();
+    assertEquals(0, ((DisposableLanguageServer) server).getExitCode());
+
+    CodeActionParams params =
+        new CodeActionParams(new TextDocumentIdentifier(DOCUMENT_URI), null, null);
+    CompletionParams completionParams =
+        new CompletionParams(new TextDocumentIdentifier(DOCUMENT_URI), new Position(0, 8));
+
+    Position position = new Position(0, 2);
+    TextDocumentIdentifier testTextDocumentIdentifier = new TextDocumentIdentifier(DOCUMENT_URI);
+    TextDocumentPositionParams positionParams =
+        new TextDocumentPositionParams(testTextDocumentIdentifier, position);
+    ReferenceParams referenceParams = new ReferenceParams();
+
+    assertEquals(SHUTDOWN_RESPONSE, service.codeAction(params).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.definition(positionParams).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.completion(completionParams).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.references(referenceParams).get()); // NOSONAR
+    assertEquals(SHUTDOWN_RESPONSE, service.documentHighlight(positionParams).get()); // NOSONAR
+    assertEquals(
```

Comment:
Assertion between inconvertible types

Reply:
These are the shutdown case where irrespective of what the request is, after shutdown, we respond with a fixed response(invalid request). i.e. response changes to a fixed string for all requests.